src/raft/raw_node.rs

```diff
@@ -46,8 +46,8 @@ pub enum SnapshotStatus {
     Failure,
 }
 
-fn is_local_msg(m: &Message) -> bool {
-    match m.get_msg_type() {
+pub fn is_local_msg(t: MessageType) -> bool {
```

Comment:
1. The existing test tests are in `tests`. It's good to keep new case companied with them.
2. It has limits to put test code in the same source file. There seems to no good way to share some common code. E.g. there are duplicated `new_entry()` in raft module for tests.